builder/oracle/classic/builder.go

```diff
@@ -0,0 +1,107 @@
+package classic
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/go-cleanhttp"
+	"github.com/hashicorp/go-oracle-terraform/compute"
+	"github.com/hashicorp/go-oracle-terraform/opc"
+	ocommon "github.com/hashicorp/packer/builder/oracle/common"
+	"github.com/hashicorp/packer/common"
+	"github.com/hashicorp/packer/helper/communicator"
+	"github.com/hashicorp/packer/packer"
+	"github.com/mitchellh/multistep"
+)
+
+// BuilderId uniquely identifies the builder
+const BuilderId = "packer.oracle.classic"
+
+// Builder is a builder implementation that creates Oracle OCI custom images.
+type Builder struct {
+	config *Config
+	runner multistep.Runner
+}
+
+func (b *Builder) Prepare(rawConfig ...interface{}) ([]string, error) {
+	config, err := NewConfig(rawConfig...)
+	if err != nil {
+		return nil, err
+	}
+	b.config = config
```

Comment:
validate that required config is present

Reply:
Should I do this here or inside of the NewConfig function?

Reply:
inside NewConfig, since I think it's good to keep it all together.

Reply:
will this now support sending a --user-data-file?in the same way, as you can do so in AWS?  It is now a feature of the oci command line program?   If you are using the standard Oracle windows severe 2012 images you need to be able to send a "setup" Powershell /batch file to make sure the windows rm and firewalls are configured correctly so that packer can build, especially when packer may be in one Oracle datacenter and you are building in another?

Reply:
This is an MVP of the builder and does not yet have that option. Would you please open a new issue with your feature request?